gnosis/src/main/scala/edu/uci/ics/cloudberry/gnosis/NewYorkGnosis.scala

```diff
@@ -0,0 +1,66 @@
+package edu.uci.ics.cloudberry.gnosis
+
+import java.io.{File, FilenameFilter}
+
+import com.vividsolutions.jts.geom.{Coordinate, Envelope, Geometry}
+import play.api.libs.json.{JsObject, Json, Writes}
+
+class NewYorkGnosis(levelGeoPathMap: Map[TypeLevel, File]) {
+
+  import NewYorkGnosis._
+
+  val levelShapeMap: Map[TypeLevel, NYGeoJSONIndex] = load(levelGeoPathMap)
+
+  private def load(shapeMap: Map[TypeLevel, File]): Map[TypeLevel, NYGeoJSONIndex] = {
+    NYLevels.map(level => {
+      val index = new NYGeoJSONIndex()
+      loadShape(shapeMap.get(level).get, index)(INYGeoJSONEntity.apply)
+      level -> index
+    }).toMap
+  }
+
+  lazy val neighbors: Seq[NYNeighborEntity] = {
+    levelShapeMap.get(NeighborLevel).get.entities.map(_.asInstanceOf[NYNeighborEntity])
+  }
+
+  lazy val neighborShapes: IGeoIndex = levelShapeMap.get(NeighborLevel).get
+
+  // used in geo tag
+  def tagPoint(longitude: Double, latitude: Double): Option[NYGeoTagInfo] = {
+    val box = new Envelope(new Coordinate(longitude, latitude))
+    neighborShapes.search(box).headOption.map(entity => NYGeoTagInfo(entity.asInstanceOf[NYNeighborEntity]))
+  }
+
```

Comment:
we also need a similar `tagRectangle` method to answer the query from the frontend which give us a rectangle and answer it with a list of neighbors/boro.